scripts/coverage.sh

```diff
@@ -32,12 +33,21 @@ generate_cover_data() {
 
     for pkg in "$@"; do
         f="$workdir/$(echo $pkg | tr / -).cover"
-        go test -v -covermode="$mode" -coverprofile="$f" "$pkg"
+        # go test -v -covermode="$mode" -coverprofile="$f" "$pkg"
+        # not using -v verbose flag, too much info to parse easily; output to file
+        go test -covermode="$mode" -coverprofile="$f" "$pkg" >> "$report"
     done
 
     echo "mode: $mode" >"$profile"
     grep -h -v "^mode:" "$workdir"/*.cover | $dir/exclude_ignore.py | egrep -v "$excludes" >>"$profile"
 }
 
-generate_cover_data $(go list ./... | grep -v /vendor/ | grep -v integration )
+generate_cover_data $(go list ./pkg/* | grep -v /vendor/ | grep -v integration )
```

Comment:
dispatchcli is under pkg, so it should be included in this rule. Unless you meant dispatch/cmd, but those are fine to skip since there is not much logic in them.

Reply:
What I meant was we will be missing coverage for dispatch/pkg/dispatchcli/cmd since ``go list ./pkg/*`` won't list it

```
dispatch git:(feb_add_oidc_provider) ✗ go list ./pkg/*
can't load package: package github.com/vmware/dispatch/pkg/dispatchcli: no Go files in /Users/smurugesan/workspace/src/github.com/vmware/dispatch/pkg/dispatchcli
```

Reply:
Good catch. Also, `go test ./pkg/api-manager` won't execute test in `./pkg/api-manager/gateway/kong/`. Looks like we need `go list -e ./pkg/*` combined with `go test $pkg/...`, but I'm not sure if this will play nicely with coverage. @zimengyang can you try that?

Reply:
I just tried "go test" with various options. Currently, Go 1.9.2 go test doesn't support multiple packages. So go test $pkg/... doesn't work in our case. In Go 1.10 release notes, they are saying go test can work with multiple packages now. If we are upgrading Go to 1.10, I will have another try.